grakn-graql/src/test/java/ai/grakn/graql/internal/gremlin/fragment/DirectIsaTest.java

```diff
@@ -0,0 +1,222 @@
+        assertTrue(plan.get(6) instanceof OutRolePlayerFragment); // go to the other role player
+        assertTrue(plan.get(7) instanceof NeqFragment); // check two players are different
+        assertTrue(plan.get(8) instanceof OutIsaFragment); // check the role player's type
+
+        // test type with subtypes
+
+        pattern = x.directIsa(thingy);
+        plan = getPlan(pattern);
+        assertEquals(2, plan.size());
+        assertTrue(plan.get(0) instanceof LabelFragment);
+        assertTrue(plan.get(1) instanceof InIsaFragment);
+
+        pattern = x.isa(thingy);
+        plan = getPlan(pattern);
+
+        assertEquals(3, plan.size());
+        assertTrue(plan.get(0) instanceof LabelFragment);
+        assertTrue(plan.get(1) instanceof InSubFragment);
+        assertTrue(plan.get(2) instanceof InIsaFragment);
+
+        pattern = and(
+                x.directIsa(thingy),
+                y.directIsa(thingy),
+                var().rel(x).rel(y).isa(related));
+        plan = getPlan(pattern);
+
+        assertEquals(9, plan.size());
+        // 3 labels: thingy1, thingy1 and related
+        assertTrue(plan.get(0) instanceof LabelFragment);
+        assertTrue(plan.get(1) instanceof LabelFragment);
+        assertTrue(plan.get(2) instanceof LabelFragment);
+        assertTrue(plan.get(3) instanceof InIsaFragment); // start from relationship type
+        assertTrue(plan.get(4) instanceof OutRolePlayerFragment); // go to a role player
+        assertTrue(plan.get(5) instanceof OutIsaFragment); // check the role player's type
+        assertTrue(plan.get(6) instanceof OutRolePlayerFragment); // go to the other role player
+        assertTrue(plan.get(7) instanceof NeqFragment); // check two players are different
+        assertTrue(plan.get(8) instanceof OutIsaFragment); // check the role player's type
+
+        // combine isa and direct isa
+
+        pattern = and(
+                x.isa(thingy),
+                y.directIsa(thingy),
+                var().rel(x).rel(y).isa(related));
+        plan = getPlan(pattern);
+
+        assertEquals(10, plan.size());
+        // 3 labels: thingy, thingy and related
+        assertTrue(plan.get(0) instanceof LabelFragment);
+        assertTrue(plan.get(1) instanceof LabelFragment);
+        assertTrue(plan.get(2) instanceof LabelFragment);
+        assertTrue(plan.get(3) instanceof InIsaFragment); // start from relationship type
+        assertTrue(plan.get(4) instanceof OutRolePlayerFragment); // go to a role player with direct isa
+        assertTrue(plan.get(5) instanceof OutIsaFragment); // check the role player's type
+        assertTrue(plan.get(6) instanceof OutRolePlayerFragment); // go to the other role player
+        assertTrue(plan.get(7) instanceof NeqFragment); // check two players are different
+        assertTrue(plan.get(8) instanceof OutIsaFragment); // check the role player's type
+        assertTrue(plan.get(9) instanceof OutSubFragment); // check the subtypes
```

Comment:
There's a method `Matchers.instanceOf` that will help you for this:

```
assertThat(plan, contains(
    instanceOf(LabelFragment.class),
    instanceOf(LabelFragment.class),
    instanceOf(LabelFragment.class),
    instanceOf(InIsaFragment.class),
    instanceOf(OutRolePlayerFragment.class),
    instanceOf(OutIsaFragment.class),
    instanceOf(OutRolePlayerFragment.class),
    instanceOf(NeqFragment.class),
    instanceOf(OutIsaFragment.class),
    instanceOf(OutSubFragment.class)
));
```

This will also give you nicer error messages if it goes wrong.

Reply:
what's contains?